_posts/2020-01-30-get-started-with-java-openjdk11.md

```diff
@@ -0,0 +1,497 @@
+
+    implementation 'com.squareup.okhttp3:okhttp:3.10.0'
+    implementation 'com.squareup.okio:okio:1.14.1'
+}
+```
+
+Let's see the new code for the Handler.java file.
+
+First we need to get the "query-string" from the `IRequest` in the function handler, the interface looks like this:
+
+```java
+public interface IRequest {
+    String getBody();
+    Map<String, String> getHeaders();
+    String getHeader(String key);
+    String getQueryRaw();
+    Map<String, String> getQuery();
+    String getPathRaw();
+    Map<String, String> getPath();
+}
+```
+
+In the URL we will pass a querystring of `?q=phrase` for our search, so let's see what that looks like:
+
+```java
+package com.openfaas.function;
+
+import com.openfaas.model.IHandler;
+import com.openfaas.model.IResponse;
+import com.openfaas.model.IRequest;
+import com.openfaas.model.Response;
+import java.util.Map;
+
+import java.io.IOException;
+
+import okhttp3.OkHttpClient;
+
+public class Handler implements IHandler {
+
+    public IResponse Handle(IRequest req) {
+        IResponse res = new Response();
+
+        try {
+            OkHttpClient client = new OkHttpClient();
+
+            Map<String, String> query = req.getQuery();
+            String q = query.get("q");
+
+            String url = "https://quote-garden.herokuapp.com/quotes/search/" + q;
+            okhttp3.Request request = new okhttp3.Request.Builder()
+                .url(url)
+                .build();
+
+            okhttp3.Response response = client.newCall(request).execute();
+            String ret = response.body().string();
+            res.setBody(ret);
+
+        } catch(Exception e) {
+            e.printStackTrace();
+            res.setBody(e.toString());
+        }
+
+        return res;
+    }
+}
+```
+
+Now we can run a build and deploy our function, but first let's enable Docker's new Buildkit container builder which can dramatically reduce the time taken to build images.
+
+```sh
+export DOCKER_BUILDKIT=1
+```
+
+```sh
+faas-cli up -f find-a-quote.yml
+```
+
+My build took around 15s to complete on a modest Intel i7 micro-PC.
+
+You'll be given a URL and Kubernetes will have already pulled the image and started a Pod for your function.
+
+```
+kubectl get deploy -n openfaas-fn -o wide
+```
+
+```sh
+curl -sSLf http://192.168.0.26:31112/function/find-a-quote?q=tree | jq
+```
+
+In the example above I used the `jq` utility to format the output, it looks like there were 10 results for "tree".
+
+```json
+{
+  "count": 18,
+  "results": [
+    {
+      "_id": "5d91b45d9980192a317c8acc",
+      "quoteText": "Notice that the stiffest tree is most easily cracked, while the bamboo or willow survives by bending with the wind.",
+      "quoteAuthor": "Bruce Lee"
+    },
+    {
+      "_id": "5d91b45d9980192a317c8a62",
+      "quoteText": "Notice that the stiffest tree is most easily cracked, while the bamboo or willow survives by bending with the wind.",
+      "quoteAuthor": "Bruce Lee"
+    }
+}
```

Comment:
Missed a `]`

Reply:
Thank you 👍 I trimmed the output and that's why